pkg/ha/ha.go

```diff
@@ -0,0 +1,67 @@
+package ha
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/replicatedhq/kots/pkg/k8sutil"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/utils/pointer"
+)
+
+const (
+	KUBERNETES_HA_MIN_NODE_COUNT = 3
+	RQLITE_HA_NODE_COUNT         = 3
+	REASON_NOT_ENOUGH_NODES      = "not enough nodes to run in HA mode"
+)
+
+func CanRunHA(ctx context.Context, clientset kubernetes.Interface) (bool, string, error) {
+	nodes, err := clientset.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
```

Comment:
Should we add selectors to match out node affinities? https://github.com/replicatedhq/kots/blob/10b6d379ae2b7ade0e482e3ebbb9fe22709da00e/pkg/kotsadm/objects/affinity_objects.go#L8